pkg/api/v1/types.go

```diff
@@ -2026,16 +2026,16 @@ type NodeSelector struct {
 // A null or empty node selector term matches no objects.
 type NodeSelectorTerm struct {
 	//Required. A list of node selector requirements. The requirements are ANDed.
-	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" protobuf:"bytes,1,rep,name=matchExpressions"`
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	MatchExpressions []NodeSelectorRequirement `json:"matchExpressions" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,rep,name=matchExpressions"`
```

Comment:
has this field shipped in a previous release this way? if so, clients are computing patches without a merge strategy (meaning they expect the server to replace the contents of the slice with whatever is given), and servers are applying patches without a merge strategy

a new client with this merge strategy would compute a patch that would only send a changed item to the server. if that patch were sent to an old server, the old server would replace the entire slice with the patch content (e.g. drop all the unchanged items)

an old client without this merge strategy would send the full contents of the slice if any item changed, not compute removals. if that patch were sent to a new server, the new server would merge in the items in the slice to the existing slice (e.g. make it impossible to remove items from the slice by patching).